client/my-sites/site-monitoring/components/site-logs-table/index.tsx

```diff
@@ -51,7 +51,7 @@ export const SiteLogsTable = memo( function SiteLogsTable( {
 		return generateRowKeys( logs );
 	}, [ logs ] );
 
-	const [ latestLogType, setLatestLogType ] = useState( logType );
+	const [ latestLogType, setLatestLogType ] = useState< LogType | undefined | null >( null );
```

Comment:
Because I'm a TypeScript/JavaScript newbie, how does this fix the issue?

Reply:
*Context*:
The log tables have two different loading states.
One that is a skeleton when it loads the first time or when you change between tabs.
And the second one which is a freeze-opacity when you change the page in the same table.

*Problem*:
The approach that I ported from Philip is that when you change between logs tab `logType`, we display the skeleton version of loading. We use `latestLogType` to track what was the previous `logType` value.

But when you came from the charts tab or when you were loading the logs directly. The component was initializing the `latestLogType` to the current `logType` in the first render which was skipping the loading state.

*Solution*:
To fix the problem, I changed the initial value of `latestLogType` to ` useState( null )`. So we know we are mounting the component for the first time and we should display the skeleton.

The `<`, `>` on `useState< LogType | undefined | null >( null )` are used to define the type for the `latestLogType` variable.

Reply:
The simplest solutions are my favorite ones ❤️